core/src/main/java/org/mule/component/DefaultComponentLifecycleAdapter.java

```diff
@@ -339,7 +338,7 @@ public Object invoke(MuleEvent event) throws MuleException
         {
             if (componentObject == null)
             {
-                throw new ComponentException(createStaticMessage("componentObject is null"), RequestContext.getEvent(), component);
+                throw new ComponentException(createStaticMessage("componentObject is null"), event, component);
```

Comment:
I think that is not correct to use the event. That could fix this particular issue, but suppose that in the exception strategy there is another component that uses Requestcontext.getEvent... that event will be incorrect.